FetchMigration/python/tests/test_endpoint_utils.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import pickle
+import random
+import unittest
+from typing import Optional
+from unittest.mock import MagicMock, patch
+
+from moto import mock_iam
+
+import endpoint_utils
+from tests import test_constants
+
+# Constants
+SUPPORTED_ENDPOINTS = ["opensearch", "elasticsearch"]
+INSECURE_KEY = "insecure"
+CONNECTION_KEY = "connection"
+TEST_KEY = "test_key"
+BASE_CONFIG_SECTION = {
+    TEST_KEY: [{"invalid_plugin1": {"key": "val"}}, {"invalid_plugin2": {}}]
+}
+
+
+# Utility method to create a test plugin config
+def create_plugin_config(host_list: list[str],
+                         basic_auth_tuple: Optional[tuple[Optional[str], Optional[str]]] = None,
+                         aws_config_snippet: Optional[dict] = None,
+                         disable_auth: Optional[bool] = None
+                         ) -> dict:
+    config = dict()
+    config["hosts"] = host_list
+    if disable_auth:
+        config["disable_authentication"] = disable_auth
+    elif basic_auth_tuple:
+        user, password = basic_auth_tuple
+        if user:
+            config["username"] = user
+        if password:
+            config["password"] = password
+    elif aws_config_snippet:
+        config.update(aws_config_snippet)
+    return config
+
+
+# Utility method to creat a test config section
+def create_config_section(plugin_config: dict) -> dict:
+    valid_plugin = dict()
+    valid_plugin[random.choice(SUPPORTED_ENDPOINTS)] = plugin_config
+    config_section = copy.deepcopy(BASE_CONFIG_SECTION)
+    config_section[TEST_KEY].append(valid_plugin)
+    return config_section
```

Comment:
Out of curiosity, can you explain what this is doing? especially the `random.choice` part please.
NIT: missing 'e' in "create" in the comment.

Reply:
The Data Prepper pipeline configuration file is composed of several "sections" - `source` and `sink` being the two familiar ones. Each section can be configured with one or more plugins. Every plugin has a name, and an associated plugin configuration.

This utility method helps in the creation of one of those "sections" for testing. The `random.choice` logic randomly picks a supported endpoint as the plugin name, and uses the supplied `plugin_config` argument as its plugin configuration value. Instead of `source` or `sink`, the section name uses the value of `TEST_KEY`

Fixed the typo as a part of the latest commit.

Reply:
I'm going to go ahead and merge this. @okhasawn if you have any other comments, please continue to provide them and i'll incorporate them in a follow-up PR.